pkg/vmprovider/providers/vsphere2/vmprovider_vm_utils.go

```diff
@@ -85,8 +86,16 @@ func GetVirtualMachineImageSpecAndStatus(
 			return nil, nil, nil, fmt.Errorf("%s: %w", msg, err)
 		}
 
+		if err := kube.SyncGVKToObject(clusterVMImage, k8sClient.Scheme()); err != nil {
```

Comment:
For my understanding, this is only needed because we are referencing the GVK from the resource later in the reconcile, right?   If yes, I think a comment stating that will help readers.